src/api/response.h

```diff
@@ -0,0 +1,77 @@
+#ifndef API___RESPONSE_H
+#define API___RESPONSE_H
+
+#include <string>  // std::string
+#include <utility> // std::pair
+#include <vector>  // std::vector
+
+using Header = std::pair<std::string, std::string>;
+using Headers = std::vector<Header>;
+
+// Represents an HTTP response.
+//
+// Usage:
+//   Response r;
+//   r.SetStatus(OK);
+//   r.SetBody(...);
+//   return r.ToString();
+//
+// Constructed by the RequestHandler, after which the server should call ToString
+// to serialize.
+class Response {
+public:
+    Response() : version_("HTTP/1.1") {}
+    ~Response() = default;
+
+    enum ResponseCode {
+        HTTP_200_OK = 200,
+        HTTP_302_FOUND = 302,
+        HTTP_404_NOT_FOUND = 404
+    };
+    
+    // Simple accesors
+    std::string version() { return version_; }
+    ResponseCode status() { return status_; }
+    std::string status_message() { return status_message_; }
+    Headers headers() { return headers_; }
+    std::string body() { return body_; }
```

Comment:
for some reason the accessors were lower case but mutators had that weird camel case. Those functions came straight from the API so I just let them be.